instance/models/load_balancer.py

```diff
@@ -111,6 +145,55 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.domain
 
+    def modifies_configuration(method):  # nopep8 # pylint: disable=no-self-argument
+        """
+        Flag an internal method as one that modifies the state of load balancer configuration,
+        to allow managing configuration state outside and around the method call.
+        """
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            """
+            Used to wrap the function that modified load balancer configuration state.
+            """
+            try:
+                # Prevent race conditions by updating state only if the LB is actually clean or dirty.
```

Comment:
For the curious, this is [optimistic locking](https://en.wikipedia.org/wiki/Optimistic_concurrency_control), so we can avoid [pessimistic locking](https://en.wikipedia.org/wiki/Concurrency_control#Categories), which I found a lot harder to implement, and harder to understand/read in the code.